eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Plutus/Evaluate.hs

```diff
@@ -0,0 +1,214 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Cardano.Ledger.Alonzo.Plutus.Evaluate (
+  evalPlutusScripts,
+  evalPlutusScriptsWithLogs,
+  CollectError (..),
+  collectPlutusScriptsWithContext,
+  lookupPlutusScript,
+)
+where
+
+import Cardano.Ledger.Alonzo.Core hiding (TranslationError)
+import Cardano.Ledger.Alonzo.Plutus.Context (EraPlutusContext (..))
+import Cardano.Ledger.Alonzo.Scripts (AlonzoEraScript (..), plutusScriptLanguage)
+import Cardano.Ledger.Alonzo.Tx (ScriptPurpose (..), indexedRdmrs)
+import Cardano.Ledger.Alonzo.TxWits (AlonzoEraTxWits (..))
+import Cardano.Ledger.Alonzo.UTxO (AlonzoEraUTxO (getSpendingDatum), AlonzoScriptsNeeded (..))
+import Cardano.Ledger.BaseTypes (ProtVer (pvMajor), natVersion)
+import Cardano.Ledger.Binary (DecCBOR (..), EncCBOR (..))
+import Cardano.Ledger.Binary.Coders
+import Cardano.Ledger.Plutus.CostModels (CostModels (..))
+import Cardano.Ledger.Plutus.Data (getPlutusData)
+import Cardano.Ledger.Plutus.Evaluate (
+  PlutusDatums (..),
+  PlutusWithContext (..),
+  ScriptResult (..),
+  runPlutusScriptWithLogs,
+ )
+import Cardano.Ledger.Plutus.Language (Language (..))
+import Cardano.Ledger.UTxO (EraUTxO (..), ScriptsProvided (..), UTxO (..))
+import Cardano.Slotting.EpochInfo (EpochInfo)
+import Cardano.Slotting.Time (SystemStart)
+import Control.Monad (guard)
+import Data.List (intercalate)
+import qualified Data.Map.Strict as Map
+import Data.Maybe (mapMaybe)
+import qualified Data.Set as Set
+import Data.Text (Text)
+import Debug.Trace (traceEvent)
+import GHC.Generics
+import Lens.Micro
+import NoThunks.Class (NoThunks)
+
+-- ===============================================================
+-- From the specification, Figure 8 "Scripts and their Arguments"
+-- ===============================================================
```

Comment:
Out of curiosity, I checked the alonzo spec, and Figure 8 is now something else: 

![2023-12-12_17-14](https://github.com/input-output-hk/cardano-ledger/assets/2070003/cf630748-a05f-43ab-b06a-55d53bc2828e)

Figure 7 is "Scripts and their arguments", but don't know, this comment seems pretty unnecessary to me , since the implementation is quite different.
Here is Figure 7:

![2023-12-12_17-14_1](https://github.com/input-output-hk/cardano-ledger/assets/2070003/0224adfb-1c8a-4aa4-a442-7f03e27b1272)

Reply:
Looks like the comment has off-by-one error :slightly_smiling_face: 

It should say:
```
 From the specification, Figure 7 "Scripts and their Arguments"
```

I'll fix it my subsequent PR.